Automattic/jetpack
# CRM: Webpack - build additional non-minified CSS for production

## Proposed changes:

* This PR includes changes to `webpack.config.js` to allow for non minified CSS files to be generated.
* With the previous set up, in a development build the CssMinimizerPlugin was actually generating the compiled CSS and not minifying it either, but renaming CSS file extensions to be `.min.css`. In production it was minifying the generated CSS (but not leaving a CSS copy).
* The new changes allow for CSS files (`.css`) to be generated in both instances (development or production).
* The PR also includes a modification to the `package.json` clean script to clean up non minified `.rtl.css` files that are now being generated (they weren't before) as well as `.map` files, plus adding those `.rtl.css` file extensions to `.gitignore`.
* This PR also includes slight changes to several CSS files - an extra blank line at the end (the result when compiling the CSS files from the SCSS files). Any other changes to SCSS files since the monorepo move will also be included here within the generated CSS files.
* I've also removed sourcemaps (from development - previously only excluded from production), as the CSS files in production builds locally would then differ from development builds. It seems the best options here would be either to remove sourcemaps altogether as I've done here, or add them in production too.
* Note - the current changes will mean we're committing changes to non-minified CSS to the repo as we already had CSS files there (removing them is also an option). 


### Other information:


## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/2928

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* On a local Jetpack development site, make a CSS change to a SCSS file that will ordinarily be compiled and minified (`scss/JetpackCRM.admin.resources-page.scss` for example). I tend to change something at the end of the file so that I can easily check for the change on files that should have been minified as well.
* Run `jetpack build plugins/crm --production` and you should find the CSS file has been modified to include the change (in the CSS directory), and the `.min.css` still includes the change. 
* Run `jetpack build plugins/crm` as you would normally when developing - the changes to the CSS file in particular should be the same.
